### ISSUE TITLE
MinGW/MSYS: Gambit is usable only as interpreter

### ISSUE THREAD

**Issue by ivsukin** [gambit/gambit#828]
**Environment:**
Freshly installed MinGW(UCRT or MinGW32 or MinGW64)/MSYS with needed development packages (gcc, autotools, make, etc.)

**Steps to reproduce:**
1. Download the latest Gambit sources, unpack them and cd into them.
2. ./configure <list of recommended options>
3. make && make check && make install
4. Create a file _test.scm_ (contents are something simple such _(display "Hello, world!")_) anywhere on the system. 
5. Run `gsc -exe test.scm`

**Expected result:**
Compiled executable as it was on some previous releases of Gambit.

**Actual result:**
`
$ gsc -exe test.scm
*** ERROR IN ##gambuild -- No such file or directory
(current-directory "~~bin")
`

Also all of the scripts in the installation prefix's bin directory are *.bat, despite compilation was done under MSYS with working bash.

**Comment by feeley** [comment]
Could you report the result of

```
$ gsc -e '(pp (map path-expand (list "~~" "~~bin" "~~lib")))'
$ gsc -e '(pp (##os-path-gambitdir))'
```

I suspect the Gambit installation directory hasn't been set properly.

**Comment by ivsukin** [comment]
Yes, the paths are wrong (and they are encoded as Windows paths):
```
$ gsc -e '(pp (map path-expand (list "~~" "~~bin" "~~lib")))'
("C:\\\\ucrt64\\" "C:\\\\ucrt64\\bin\\" "C:\\\\ucrt64\\lib\\")

$ gsc -e '(pp (##os-path-gambitdir))'
"\\\\ucrt64"
```

I have found that the latest version of Gambit that has the correct out-of-the-box gsc behavior is 4.7.5 (although old versions of Gambit segfault at exit, but I think that this is not relevant):
```
$ gambit/v4.7.6/bin/gsc -e '(pp (map path-expand (list "~~" "~~bin" "~~lib")))'
("\\home\\user\\gambit\\v4.7.6\\"
 "\\home\\user\\gambit\\v4.7.6\\bin\\"
 "\\home\\user\\gambit\\v4.7.6\\lib\\")
Segmentation fault

$ gambit/v4.7.5/bin/gsc -e '(pp (map path-expand (list "~~" "~~bin" "~~lib")))'
("C:/msys64/home/user/gambit/v4.7.5\\"
 "C:/msys64/home/user/gambit/v4.7.5\\bin\\"
 "C:/msys64/home/user/gambit/v4.7.5\\lib\\")
Segmentation fault
```

**Comment by feeley** [comment]
I'll look into this some more. In the meantime you might try to force the installation directory location with the `-:=<dir>` runtime option like this:

```
$ gsc -:=C:/where/gambit/was/installed ...
```

I'm not sure why `gsc` thinks the installation directory is `\\ucrt64`.

**Comment by feeley** [comment]
Try this... in the file `configure.ac` remove or comment out the `PATH_SEP=...` line here:

```
    case "$target_os" in

       mingw* | msys*)

               # add Windows libraries needed by MinGW

               PATH_SEP="\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"
```

Then run `autoconf` then `./configure` then `make`.

I'm sorry but I can't test this locally because I don't have easy access to a Windows machine at the moment...

**Comment by ivsukin** [comment]
Thank you, now it works as expected:
```
$ gambit/v4.9.4/bin/gsc -e '(pp (map path-expand (list "~~" "~~bin" "~~lib")))'
("C:/msys64/home/user/gambit/v4.9.4\\"
 "C:/msys64/home/user/gambit/v4.9.4\\bin\\"
 "C:/msys64/home/user/gambit/v4.9.4\\lib\\")

$ gambit/v4.9.4/bin/gsc -e '(pp (##os-path-gambitdir))'
"C:/msys64/home/user/gambit/v4.9.4"
```
and
```
$ cat test.scm
(display "Hello world!")
(newline)

$ gambit/v4.9.4/bin/gsc -exe test.scm

$ ./test
Hello world!
```

**Comment by ivsukin** [comment]
Is there any way to force Gambit installation system to produce bash versions of scripts on MSYS?

**Comment by feeley** [comment]
I haven't tested this, but it is probably as simple as modifying the `configure.ac` script around these lines:

```
exe="$EXEEXT"
obj=".$OBJEXT"

if test "$exe" = ".exe"; then
  bat=".bat"
else
  bat=""
fi
```

just force `bat=""`.  The value of `bat` is the extension for shell scripts.  The makefiles test this to know if a POSIX shell script or a Windows batch file must be created.
